storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -0,0 +1,604 @@
+                }
+            }
+        }
+        return stormBases;
+    }
+
+    public static void validateDistributedMode(Map conf) {
+        if (ConfigUtils.isLocalMode(conf)) {
+            throw new IllegalArgumentException("Cannot start server in local mode!");
+        }
+    }
+
+    private static void validateIds(StormTopology topology) throws InvalidTopologyException {
+        List<String> componentIds = new ArrayList<String>();
+
+        for (StormTopology._Fields field : Thrift.getTopologyFields()) {
+            if (ThriftTopologyUtils.isWorkerHook(field) == false) {
+                Object value = topology.getFieldValue(field);
+                if (value != null) {
+                    Map<String, Object> componentMap = (Map<String, Object>) value;
+                    componentIds.addAll(componentMap.keySet());
+
+                    for (String id : componentMap.keySet()) {
+                        if (Utils.isSystemId(id)) {
+                            throw new InvalidTopologyException(id + " is not a valid component id.");
+                        }
+                    }
+                    for (Object componentObj : componentMap.values()) {
+                        ComponentCommon common = getComponentCommon(componentObj);
+                        Set<String> streamIds = common.get_streams().keySet();
+                        for (String id : streamIds) {
+                            if (Utils.isSystemId(id)) {
+                                throw new InvalidTopologyException(id + " is not a valid stream id.");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        List<String> offending = Utils.getRepeat(componentIds);
+        if (offending.isEmpty() == false) {
+            throw new InvalidTopologyException("Duplicate component ids: " + offending);
+        }
+    }
+
+    private static boolean isEmptyInputs(ComponentCommon common) {
+        if (common == null) {
+            return true;
+        } else if (common.get_inputs() == null) {
+            return true;
+        } else {
+            return common.get_inputs().isEmpty();
+        }
+    }
+
+    public static Map<String, Object> allComponents(StormTopology topology) {
+        Map<String, Object> components = new HashMap<String, Object>();
+        List<StormTopology._Fields> topologyFields = Arrays.asList(Thrift.getTopologyFields());
+        for (StormTopology._Fields field : topologyFields) {
+            if (ThriftTopologyUtils.isWorkerHook(field) == false) {
+                components.putAll(((Map) topology.getFieldValue(field)));
+            }
+        }
+        return components;
+    }
+
+    public static Map componentConf(Object component) {
+        Map<Object, Object> conf = new HashMap<Object, Object>();
+        ComponentCommon common = getComponentCommon(component);
+        if (common != null) {
```

Comment:
The original code didn't check for null and would throw an NPE if we passed in a null.  We should do the same.